### PR TITLE
add equality check for intervals

### PR DESCRIPTION
**Pull request by ishmum123**
This would also be needed for #133

### CODE REVIEW

File: dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedIntervalTest.java [hunk 1]
```diff
@@ -36,7 +37,37 @@ public class ZonedIntervalTest {
   private final ZonedDateTime ONE_DAY_AGO = getTimeSubtractingDays(1);
 
   @Test
-  public void testInclusiveEndIsSmallerThanExclusiveEnd() {
+  public void test_equals_trueWhenStartAndEndTimeAreEqual() {
```
Comment (dawidxc) [inline dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedIntervalTest.java:40]: Optional: I would suggest to drop the `test_` prefix.

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedInterval.java [hunk 2]
```diff
@@ -82,4 +82,12 @@ public ZonedInterval span(@Nonnull ZonedInterval interval) {
   public String toString() {
     return "[" + getStart() + ".." + getEndExclusive() + ")";
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof ZonedInterval)) return false;
+    ZonedInterval interval = (ZonedInterval) obj;
+    return interval.getStart().isEqual(getStart())
```
Comment (dawidxc) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedInterval.java:90]: nit: In the same class the fields can be accessed directly, i.e. `interval.start.isEqualTo(start)`.

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedInterval.java [hunk 3]
```diff
@@ -82,4 +82,12 @@ public ZonedInterval span(@Nonnull ZonedInterval interval) {
   public String toString() {
     return "[" + getStart() + ".." + getEndExclusive() + ")";
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof ZonedInterval)) return false;
+    ZonedInterval interval = (ZonedInterval) obj;
```
Comment (dawidxc) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedInterval.java:89]: nit: I would suggest to rename to `otherInterval`.

**Review by dawidxc — APPROVED** [review]

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedInterval.java [hunk 4]
```diff
@@ -82,4 +82,11 @@ public ZonedInterval span(@Nonnull ZonedInterval interval) {
   public String toString() {
     return "[" + getStart() + ".." + getEndExclusive() + ")";
   }
+
+  @Override
+  public boolean equals(Object obj) {
```
Comment (paolomorandini) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedInterval.java:87]: Please also implement `hashCode`.

**Review by paolomorandini — CHANGES_REQUESTED** [review]

**Review by paolomorandini — APPROVED** [review]

**Pull request merged**